src/fetching/getTodoEvents.ts

```diff
@@ -0,0 +1,34 @@
+import { defaultFetch } from '.';
+import { ZetkinEvent, ZetkinEventResponse } from '../types/zetkin';
+
+export default function getTodoEvents(fetch = defaultFetch) {
+    return async () : Promise<ZetkinEvent[]> => {
+
+        const membershipsRes = await fetch(`/users/me/memberships`);
+        const membershipsData = await membershipsRes.json();
+
+        const responsesRes = await fetch('/users/me/action_responses');
+        const responsesData = await responsesRes.json();
+
+        const todoEvents = [];
+
+        if (responsesData.data) {
+            for (const mObj of membershipsData.data) {
+                const oEventsRes = await fetch(`/orgs/${mObj.organization.id}/actions`);
+                const oEventsData = await oEventsRes.json();
+
+                const org = {
+                    id: mObj.organization.id,
+                    title: mObj.organization.title,
+                };
+
+                for (const eObj of oEventsData.data) {
+                    if (responsesData.data.find((r : ZetkinEventResponse) => r.action_id  === eObj.id)) {
+                        todoEvents.push({ ...eObj, organization: org });
+                    }
+                }
+            }
```

Comment:
You're looping over the full list of events, and looking to see if every event is amongst the ones to which the user responded. Lets do a thought experiment.

There are 100 events published
The user has signed up for 2 events
The two events are the middle ones (indexes 49 and 50) in the full list of events.

The way you currently have implemented this means that we will loop over 100 events. For every event, we search through (loop over, inside the `find()` method) 2 event responses to compare. This results in 200 comparisons.

Is there a faster way?